Equation.java

```diff
@@ -0,0 +1,54 @@
+import static java.lang.Math.*;
+/**
+ * solve the quadratic equation
+ * @author Babak Alexey
+ */
+public class Equation {
+
+   /** 
+    * check possibility of the "String -> Double" conversion
+    * @param string  the string to check,
+    * if string has only double variable  - return it
+    */
+   public static double checkString(String string) {	
+      double coeff;
+      try {
+         coeff = Double.parseDouble(string);
+      } catch (Exception e) {
+         return Double.POSITIVE_INFINITY;

```

Comment:
Why Double.POSITIVE_INFINITY? First of all user can input something like 2e2048 which will be parsed as POSITIVE_INFINITY and can masquerade real issue. Secondary this is not java way - you don't know here how exactly this case should be processed - so just throw exception to caller and let it decides how to process it